src/components/TextInput.tsx

```diff
@@ -0,0 +1,197 @@
+import React, {useEffect, useState} from "react";
+import {
+    Keyboard,
+    KeyboardTypeOptions,
+    StyleProp,
+    StyleSheet,
+    Text,
+    TextInput as SystemTextInput,
+    TextStyle,
+    TouchableOpacity,
+    View,
+    ViewStyle
+} from 'react-native';
+import {Colors, Indents, TextSize} from "../constants/styles";
+import Ionicons from "@expo/vector-icons/Ionicons";
+
+const minimumTouchableSize = 40;
+
+interface InputProps {
+    value: string,
+    required?: boolean,
+    keyboardType?: KeyboardTypeOptions,
+    autoCapitalize?: 'none' | 'sentences' | 'words' | 'characters',
+    isEditable?: boolean,
+    multiline?: boolean,
+    maxLength?: number,
+    onChange: (arg0: string) => void,
+    errorHandler?: (arg0: boolean) => void,
+    label?: string,
+    placeholder?: string,
+    isPassword?: boolean,
+    validation?: { rule: RegExp, message: string }[],
+    styles?: {
+        wrapper?: StyleProp<ViewStyle>,
+        input?: StyleProp<TextStyle>,
+        label?: StyleProp<TextStyle>,
+        error?: StyleProp<TextStyle>
+    }
+}
+
+const TextInput: React.FC<InputProps> = (props) => {
+    const {
+        required = false,
+        value: initialValue,
+        keyboardType,
+        autoCapitalize = 'none',
+        isEditable = true,
+        multiline = false,
+        maxLength,
+        placeholder,
+        errorHandler,
+        isPassword,
+        validation,
+        onChange,
+        styles,
+        label
+    } = props;
+
+    const [value, setValue] = useState(initialValue)
+    const [error, setError] = useState('')
+    const [editable, setEditable] = useState(isEditable)
+    const [isFocused, setFocus] = useState(false)
+    const [secureText, setSecureText] = useState(isPassword);
+
+    useEffect(() => {
+        if (required && !value) {
+            errorHandler && errorHandler(true)
```

Comment:
So you want to show error from start?

Reply:
BTW, if you give `errorHandler` default value you will not check it everywhere

Reply:
Great idea! Thanks!